### PR TITLE
Add couplingICubHandMk2 device coupling handler

### PR DESCRIPTION
**Pull request by Nicogene**
This PR adds the `couplingICubHandMk2` device driver, that implement the IJointCoupling interface:
- https://github.com/robotology/yarp/pull/3026

### CODE REVIEW

**Comment by Nicogene** [comment]
This work bring to light a thing to be understand/discussed related to devices that are outside the YARP repository but want to exploit the param parser generation feature.
In:
- #983 
- https://github.com/icub-tech-iit/ergocub-software/pull/231

We added inside the relative devices' CMakeLists:
```cmake
  option(ALLOW_DEVICE_PARAM_PARSER_GENERATION "Generate the param parser for foo device" OFF)
```
this option is forwarded in order to enable the parser generation.

@randaz81 is it the correct way the external repositories for exploiting this new feature?
Maybe we need to add this option in the root CMakeLists?

cc @traversaro @martinaxgloria

**Comment by traversaro** [comment]
I guess that we can simply leave the option not set, as we do for the similar `ALLOW_IDL_GENERATION` option?

File: src/libraries/icubmod/couplingICubHandMk2/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+|  `jointNames`                      |     -             |   list of strings       | Read / write | -       |   -           |  Yes            | Names of the physical joints                                   |                                   |
+|  `LIMITS`                      |     -             |   group      | Read / write | -       |   -           |  Yes            | group containing the physical joint limits                                     |                                   |
+|                      |     `jntPosMin`        | list of double     | Read / write | -       |   -           |  Yes            | Phyisical joints' position minimum                                  |                      |
+|                      |     `jntPosMax`        | list of double     | Read / write | -       |   -           |  Yes            | Phyisical joints' position maximum                                  |                                 |
```
[the remark this answers is not in the document]
Comment (traversaro) [reply src/libraries/icubmod/couplingICubHandMk2/README.md:25]: The typo fix is ok, but why you also changed the parameter name?

**Comment by Nicogene** [comment]
In:
- [`b371828` (#995)](https://github.com/robotology/icub-main/pull/995/commits/b371828f6ba07d169c4df9679bfda725e0a253a6)

I managed to merge all the coupling handlers (we used to have one separated per finger) into one.

I don't know who could review it (maybe @traversaro @PasMarra @xEnVrE  ?) in any case I need to give a try, in this way instead of having 10 inis (2x5 inis) for the hands we just need one ini per hand like this one:

``` ini
disableImplicitNetworkWrapper
yarpDeviceName left_hand_hardware_device

jointNames l_hand_thumb_0_joint l_hand_thumb_1_joint l_hand_thumb_2_joint l_hand_thumb_3_joint l_hand_index_0_joint l_hand_index_1_joint l_hand_index_2_joint l_hand_index_3_joint l_hand_middle_0_joint l_hand_middle_1_joint l_hand_middle_2_joint l_hand_middle_3_joint l_hand_ring_0_joint l_hand_ring_1_joint l_hand_ring_2_joint l_hand_ring_3_joint l_hand_little_0_joint l_hand_little_1_joint l_hand_little_2_joint l_hand_little_3_joint

min_stiffness 0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0
max_stiffness 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0 1000.0
min_damping   0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0
max_damping   100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0  100.0

[TRAJECTORY_GENERATION]
trajectory_type minimum_jerk

[COUPLING]
device                   couplingICubHandMk2
actuatedAxesNames        l_hand_finger l_thumb_oppose l_thumb_proximal l_thumb_distal l_index_proximal l_index_distal l_middle_proximal l_middle_distal l_pinky
actuatedAxesPosMin       0.0  0.0  0.0  0.0   0.0  0.0   0.0  0.0   0.0
actuatedAxesPosMax       60.0 90.0 90.0 180.0 90.0 180.0 90.0 180.0 270.0

[POSITION_CONTROL]
controlUnits  metric_units
controlLaw    joint_pid_gazebo_v1
kp            0.1   0.1   0.1   0.1  0.1   0.1   0.1   0.1  0.1   0.1   0.1   0.1  0.1   0.1   0.1   0.1  0.1   0.1   0.1   0.1
kd            0.01  0.01  0.01  0.01 0.01  0.01  0.01  0.01 0.01  0.01  0.01  0.01 0.01  0.01  0.01  0.01 0.01  0.01  0.01  0.01
ki            0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0
maxInt        9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999
maxOutput     9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999
shift         0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0
ko            0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0
stictionUp    0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0
stictionDwn   0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0

[VELOCITY_CONTROL]
velocityControlImplementationType integrator_and_position_pid
controlUnits  metric_units
controlLaw    joint_pid_gazebo_v1
kp            8.726 8.726 8.726 5.235 8.726 8.726 8.726 5.235 8.726 8.726 8.726 8.726 8.726 8.726 8.726 8.726 8.726 8.726 8.726 8.726
kd            0.035 0.035 0.035 0.002 0.035 0.035 0.035 0.002 0.035 0.035 0.035 0.035 0.035 0.035 0.035 0.035 0.035 0.035 0.035 0.035
ki            0.002 0.002 0.002 0.0   0.002 0.002 0.002 0.0   0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002
maxInt        9999  9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999  9999  9999  9999  9999  9999  9999  9999
maxOutput     9999  9999  9999  9999  9999 9999  9999  9999  9999 9999  9999  9999  9999  9999  9999  9999  9999  9999  9999  9999
shift         0.0   0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0
ko            0.0   0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0
stictionUp    0.0   0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0
stictionDwn   0.0   0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0

[IMPEDANCE_CONTROL]
controlUnits  metric_units
controlLaw    joint_pid_gazebo_v1
stiffness     0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0
damping       0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0 0.0   0.0   0.0   0.0

[LIMITS]
jntPosMax 0.0    0.0 20.0 20.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0 90.0
jntPosMin -20.0  0.0  0.0 0.0  0.0  0.0  0.0  0.0  0.  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  90.0 90.0
jntVelMax 100.0  100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0 100.0

```

**Comment by Nicogene** [comment]
This PR has been tested with:
- https://github.com/robotology/icub-models/pull/249
- #997 

@pattacini  @traversaro  it is ready to be reviewed/merged

**Review by traversaro — APPROVED** [review]

**Review by pattacini — APPROVED** [review]

**Pull request merged**